### ISSUE TITLE
Cannot adopt EC2 security group

### ISSUE THREAD

**Issue by mathias-muench** [aws-controllers-k8s/community#1602]
**Describe the bug**

Trying to adopt an EC2 security group as adoptedresource fails with the following status:

```
Status:
  Conditions:
    Message:  SecurityGroup.ec2.services.k8s.aws "foo" is invalid: spec.name: Required value
    Status:   False
    Type:     ACK.Adopted
Events:       <none>
```

**Steps to reproduce**

Create security group:

```
$ aws ec2 create-security-group --vpc-id vpc-************* --description foo --group-name bar
{
    "GroupId": "sg-05bcc51954336de09"
}
```

Apply manifest and describe result.

```
(main *>) $ cat foo.yaml 
apiVersion: services.k8s.aws/v1alpha1
kind: AdoptedResource
metadata:
  name: foo
spec:  
  aws:
    nameOrID: sg-05bcc51954336de09
  kubernetes:
    group: ec2.services.k8s.aws
    kind: SecurityGroup
(main *>) $ kubectl apply -f foo.yaml
adoptedresource.services.k8s.aws/foo created
(main *>) $ kubectl describe adoptedresources.services.k8s.aws
```

**Expected outcome**

Security group gets adopted.

**Environment**

* Kubernetes version: 1.22
* Using eksctl version 0.122.0
* AWS service targeted: ec2-controller v0.1.0

**Comment by RedbackThomson** [comment]
This seems to be a valid use for the `additionalKeys` part of the `AdoptedResource` spec. This section behaves differently for every API, but for the `SecurityGroup` resource it would essentially require that users provide a `groupName` key with the group name as the value. However, the code-generator wasn't able to produce an `additionalKeys` element for this field, since the associated [API call input shape](https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#DescribeSecurityGroupsInput) contains a list of strings for group name - which we [currently don't support](https://github.com/aws-controllers-k8s/code-generator/blob/39d7c30b8df3c44a9804f578cd6acc9a3f669b03/pkg/generate/code/set_resource.go#L1010-L1016).

So in short, until this limitation is lifted, we can't support adopting security groups.

**Comment by mathias-muench** [comment]
That's a pity, but I guess we'll have to live with it.
